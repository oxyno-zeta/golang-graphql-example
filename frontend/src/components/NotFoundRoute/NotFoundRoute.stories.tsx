import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import NotFoundRoute, { Props } from './NotFoundRoute';

export default {
  title: 'Components/NotFoundRoute',
  component: NotFoundRoute,
} as ComponentMeta<typeof NotFoundRoute>;

const Template: ComponentStory<typeof NotFoundRoute> = function C(args: Props) {
  return <NotFoundRoute {...args} />;
};

export const Playground = Template.bind({});
Playground.args = {};
