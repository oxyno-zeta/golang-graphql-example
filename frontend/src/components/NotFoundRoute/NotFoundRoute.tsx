import React from 'react';
import { useTranslation } from 'react-i18next';
import Typography, { TypographyProps } from '@mui/material/Typography';

export interface Props {
  typographyProps?: TypographyProps;
}

const defaultProps = {
  typographyProps: {},
};

function NotFoundRoute({ typographyProps }: Props) {
  // Setup translate
  const { t } = useTranslation();

  return (
    <Typography
      sx={{ display: 'flex', justifyContent: 'center', textAlign: 'center', margin: '10px 0' }}
      {...typographyProps}
    >
      {t('common.routeNotFound')}
    </Typography>
  );
}

NotFoundRoute.defaultProps = defaultProps;

export default NotFoundRoute;
