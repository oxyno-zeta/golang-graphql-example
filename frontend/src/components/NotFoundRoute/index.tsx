import NotFoundRoute, { Props } from './NotFoundRoute';

export default NotFoundRoute;
export type { Props };
