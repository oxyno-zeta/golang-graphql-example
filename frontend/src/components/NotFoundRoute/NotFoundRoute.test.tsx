import React from 'react';
import { render } from '@testing-library/react';
// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';

import NotFoundRoute from './NotFoundRoute';

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key: string) => key }),
}));

describe('NotFoundRoute', () => {
  it('should display no data', async () => {
    const { container } = render(<NotFoundRoute />);
    // Now find errors
    expect(container).toHaveTextContent('common.routeNotFound');
    expect(container).toMatchSnapshot();

    expect(document.title).toEqual('common.routeNotFound');
  });

  it('should display no data with specific variant', async () => {
    const { container } = render(<NotFoundRoute typographyProps={{ variant: 'body2' }} />);
    // Now find errors
    expect(container).toHaveTextContent('common.routeNotFound');
    expect(container.firstChild).toHaveClass('MuiTypography-body2');
    expect(container).toMatchSnapshot();

    expect(document.title).toEqual('common.routeNotFound');
  });
});
