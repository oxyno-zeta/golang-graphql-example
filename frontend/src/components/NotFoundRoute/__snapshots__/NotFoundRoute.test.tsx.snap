// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundRoute should display no data 1`] = `
<div>
  <p
    class="MuiTypography-root MuiTypography-body1 css-wdvp0u-MuiTypography-root"
  >
    common.routeNotFound
  </p>
</div>
`;

exports[`NotFoundRoute should display no data with specific variant 1`] = `
<div>
  <p
    class="MuiTypography-root MuiTypography-body2 css-1ngmjzi-MuiTypography-root"
  >
    common.routeNotFound
  </p>
</div>
`;
